### PR TITLE
Convert the spec for Bundler::UpdateChecker::LatestVersionFinder to use project based fixtures.

### PR DESCRIPTION
**Pull request by brrygrdn**
Follows https://github.com/dependabot/dependabot-core/pull/3370

This prepares the way for bundler version detection by ensuring we can readily produce builds against lockfiles with the correct BUNDLED WITH values based on test parameters.

### CODE REVIEW

File: bundler/spec/dependabot/bundler/update_checker/latest_version_finder_spec.rb [hunk 1]
```diff
@@ -73,6 +74,8 @@
   let(:gemfile_fixture_name) { "Gemfile" }
   let(:lockfile_fixture_name) { "Gemfile.lock" }
   let(:gemspec_fixture_name) { "example" }
```
Comment (feelepxyz) [inline bundler/spec/dependabot/bundler/update_checker/latest_version_finder_spec.rb:76]: ^^ these still used?

File: bundler/spec/dependabot/bundler/update_checker/latest_version_finder_spec.rb [hunk 2]
```diff
@@ -52,6 +52,7 @@
   let(:source) { nil }
   let(:requirement_string) { ">= 0" }
 
+  # TODO: REMOVE
```
Comment (brrygrdn) [inline bundler/spec/dependabot/bundler/update_checker/latest_version_finder_spec.rb:55]: Opps, I forgot to remove these 🤕

**Comment by brrygrdn** [comment]
Just noticed one case that slipped the net, this needs updated

File: bundler/spec/dependabot/bundler/update_checker/latest_version_finder_spec.rb [hunk 3]
```diff
@@ -439,8 +390,7 @@
     end
```
Comment (feelepxyz) [inline bundler/spec/dependabot/bundler/update_checker/latest_version_finder_spec.rb:390]: Looks like there's an example above that references the old let: https://github.com/dependabot/dependabot-core/blob/d328bdcb625fb2783bfeb2b42e6febb2617e5989/bundler/spec/dependabot/bundler/update_checker/latest_version_finder_spec.rb#L221-L223

Comment (brrygrdn) [reply bundler/spec/dependabot/bundler/update_checker/latest_version_finder_spec.rb:389]: 👍🏻  Deleted and replaced with a fixture

Comment (brrygrdn) [reply bundler/spec/dependabot/bundler/update_checker/latest_version_finder_spec.rb:389]: 👍🏻  Deleted and replaced with a fixture

**Review by feelepxyz — APPROVED** [review]
Nice one!

**Pull request merged**